E3SM-Project/E3SM
# Update CIME to ESMCI cime5.8.20

Update CIME to ESMCI cime5.8.20

Squash merge of jgfouca/branch-for-to-acme-2020-04-14

This is the first cime update in half a year, lots of changes.

Features:
* allow for specification of vertical grid in ATM_GRID
* if no env vars, print comment to assist user in env_mach_specific.xml
* case.setup: add --keep option for clean/reset actions to keep local changes to generated files
* case.setup: refresh build Macros when doing a reset/clean
* cprnc: cime will build a copy in sharelibs if no global version available
* check_input_data: add ftp timeout, make sure failover to svn is working
* added support for new OMIP compsets in POP as well as better handling of the different versions of JRA forcings
* Pio2 updated to latest master
* reorganize mct and nuopc data, xcpl and stub components based on coupler
* new auto-triaging feature for tests, will report potentially broken commits to TestStatus.log
* coupler: Allow writing T-forcing cplhist files even when running with SGLC
* Add support for UFS model! Not really relevant to E3SM, but is a major development in CIME
* archiver: Extend archiving support to accept a regex pattern
* coupler: Update for compatibility with latest CMEPS

Bug fixes:
* check_input_data: fix issue with pop input files that should not be found
* fix default for PIO_REARRANGER
* docn: Small fix to allow scam to run with aquaplanet
* fix issue with resubmit on systems that use ssh
* fix misleading output in CaseStatus
* Update description for FORCE_BUILD_SMP
* archiver: pop dd history files were not identified by get_extension
* NOTE: MPAS history files are now being processed as they should have been all along
* Improve scripts_regression_tests test cleanup
* dlnd: Some gnu tests were giving an error of different character lengths in dlnd.
* coupler: seq flux should be initialized even if for some reason the namelist isn't read
* env_batch.xml: we can now handle the case where multiple queues have the same name
* ERI: fix tests that were less than a day, or with START_TOD nonzero
* test hist: Exclude .ref1 and .ref2 from TestStatus files
* coupler: fix issue with e3sm field

Fixes: #2324 

    CONFLICTS:
      both modified:   cime/config/e3sm/config_files.xml
      both modified:   cime/scripts/lib/CIME/BuildTools/configure.py
      both modified:   cime/scripts/lib/CIME/SystemTests/system_tests_common.py
      both modified:   cime/scripts/lib/CIME/XML/env_batch.py
      both modified:   cime/scripts/lib/CIME/build.py
      both modified:   cime/scripts/lib/CIME/buildlib.py
      both modified:   cime/scripts/lib/CIME/test_scheduler.py
      both modified:   cime/scripts/lib/CIME/utils.py
      both modified:   cime/scripts/lib/CIME/wait_for_tests.py
      both modified:   cime/scripts/lib/jenkins_generic_job.py
      both modified:   cime/scripts/tests/scripts_regression_tests.py
      both modified:   cime/src/drivers/mct/cime_config/config_component.xml
      both modified:   cime/src/drivers/mct/cime_config/namelist_definition_drv.xml
      both modified:   cime/src/drivers/mct/main/cime_comp_mod.F90
      both modified:   cime/src/drivers/mct/main/prep_rof_mod.F90
